src/include/rbd/librbd.h

```diff
@@ -449,6 +450,22 @@ CEPH_RBD_API int rbd_deep_copy_with_progress(rbd_image_t image,
                                              librbd_progress_fn_t cb,
                                              void *cbdata);
 
+/* migrate */
+CEPH_RBD_API int rbd_migrate(rbd_image_t src, rados_ioctx_t dest_io_ctx,
+                             const char *destname,
+                             rbd_image_options_t dest_opts);
+CEPH_RBD_API int rbd_migrate_with_progress(rbd_image_t image,
+                                           rados_ioctx_t dest_io_ctx,
+                                           const char *destname,
+                                           rbd_image_options_t dest_opts,
+                                           librbd_progress_fn_t cb,
+                                           void *cbdata);
+CEPH_RBD_API int rbd_abort_migrate(rados_ioctx_t io, const char *name);
```

Comment:
Nit: I would name these using ```<noun>_<verb>```-style, so ```rbd_migrate_abort``` (similar comment for C++ and Python APIs)